src/main/java/tech/pegasys/net/core/tx/export/TxExportService.java

```diff
@@ -0,0 +1,140 @@
+package tech.pegasys.net.core.tx.export;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import org.tinylog.Logger;
+import org.web3j.utils.Numeric;
+import tech.pegasys.net.api.model.ActionableAccount;
+import tech.pegasys.net.api.model.EIP1559Transaction;
+import tech.pegasys.net.api.model.LegacyTransaction;
+import tech.pegasys.net.api.service.ChainFiller;
+import tech.pegasys.net.api.service.transaction.TransactionSigner;
+import tech.pegasys.net.config.ChainFillerConfiguration;
+
+public class TxExportService {
+
+  private final ChainFiller chainFiller;
+  private final ChainFillerConfiguration configuration;
+  private final List<ActionableAccount> accounts;
+  private final Path outputPath;
+
+  public TxExportService(
+      final ChainFiller chainFiller,
+      final ChainFillerConfiguration configuration,
+      final List<ActionableAccount> accounts) {
+    this.chainFiller = chainFiller;
+    this.configuration = configuration;
+    this.accounts = accounts;
+    this.outputPath = Paths.get(configuration.transactionExportFile());
+    try {
+      Files.createFile(outputPath);
+    } catch (final FileAlreadyExistsException e) {
+      Logger.warn("export file exists and will be deleted");
+      try {
+        Files.delete(outputPath);
+        Files.createFile(outputPath);
+      } catch (final IOException ioe) {
+        Logger.error(e, "cannot create export file");
+        throw new RuntimeException(e);
+      }
+    } catch (final IOException e) {
+      Logger.error(e, "cannot create export file");
+      throw new RuntimeException(e);
+    }
+  }
+
+  public void start() {
+    try {
+      final ExecutorService executorService =
+          Executors.newFixedThreadPool(configuration.numThreads());
+      final int eip1559Txs =
+          (int) Math.ceil(configuration.numTransactions() * configuration.eip1559TxWeight());
+      final int legacyTxs = configuration.numTransactions() - eip1559Txs;
+      accounts.forEach(
+          actionableAccount -> {
+            executorService.submit(
```

Comment:
Yes you are right. I will update the implementation

Reply:
Actually they are already interleaved. I did some tests. And since i submit the tasks in an executor service they are run in parallel. I will just do more tests to see if there are no race conditions to access the file. Would it be better to add all transactions to a list and once all are generated write them to the output file ?

Reply:
As long as they are interleaved I'm satisfied.

Reply:
Ok i will do more tests before merging and if you are ok i will merge if everything is working